lib/util/stream_utils.js

```diff
@@ -761,37 +761,61 @@ shaka.util.StreamUtils = class {
    * @param {shaka.extern.Manifest} manifest
    * @private
    */
-  static filterImageStreams_(manifest) {
-    // Filter image streams.
-    manifest.imageStreams = manifest.imageStreams.filter((stream) => {
-      // TODO: re-examine this and avoid allow-listing the MIME types we can
-      // accept.
-      const validMimeTypes = [
-        'image/svg+xml',
-        'image/png',
-        'image/jpeg',
-      ];
-      const Platform = shaka.util.Platform;
-      // Add webp support to popular platforms that support it.
-      const webpSupport = Platform.isWebOS() ||
-                          Platform.isTizen() ||
-                          Platform.isChromecast();
-      if (webpSupport) {
-        validMimeTypes.push('image/webp');
+  static async filterImageStreams_(manifest) {
+    const imageStreams = [];
+    for (const stream of manifest.imageStreams) {
+      const mimeType = stream.mimeType;
+      if (!shaka.util.StreamUtils.supportedImageMimeTypes.has(mimeType) &&
+          !shaka.util.StreamUtils.unsupportedImageMimeTypes.has(mimeType)) {
+        const minImage = shaka.util.StreamUtils.minImage.get(mimeType);
+        if (minImage) {
+          // eslint-disable-next-line no-await-in-loop
+          const res = await shaka.util.StreamUtils.isImageSupported_(minImage);
+          if (res) {
+            shaka.util.StreamUtils.supportedImageMimeTypes.add(mimeType);
+          } else {
+            shaka.util.StreamUtils.unsupportedImageMimeTypes.add(mimeType);
+          }
+        } else {
+          shaka.util.StreamUtils.unsupportedImageMimeTypes.add(mimeType);
+        }
       }
-      // TODO: add support to image/webp and image/avif
-      const keep = validMimeTypes.includes(stream.mimeType);
+
+      const keep = shaka.util.StreamUtils.supportedImageMimeTypes.has(mimeType);
 
       if (!keep) {
         shaka.log.debug('Dropping image stream. Is not supported by the ' +
                         'platform.', stream);
+      } else {
+        imageStreams.push(stream);
       }
+    }
+    manifest.imageStreams = imageStreams;
+  }
 
-      return keep;
+  /**
+   * @param {string} minImage
+   * @return {!Promise.<boolean>}
+   * @private
+   */
+  static isImageSupported_(minImage) {
+    return new Promise((resolve) => {
+      const imageElement = /** @type {HTMLImageElement} */(new Image());
+      imageElement.src = minImage;
+      if ('decode' in imageElement) {
+        imageElement.decode().then(() => {
```

Comment:
Please use async/await and try/catch instead of .then.catch.

Reply:
hum... I'm getting a error:
```
/Users/alvaro.velad/workspace/pullrequest-shaka-player/lib/util/stream_utils.js
  806:15  error  Parsing error: Unexpected token imageElement
```

With
```
static isImageSupported_(minImage) {
    const imageElement = /** @type {HTMLImageElement} */(new Image());
    imageElement.src = minImage;
    if ('decode' in imageElement) {
      try {
        await imageElement.decode();
        return true;
      } catch (e) {
        return false;
      }
    } else {
      return new Promise(async (resolve) => {
        imageElement.onload = imageElement.onerror = () => {
          resolve(imageElement.height === 2);
        };
      });
    }
  }
```

Reply:
@joeyparrish can you help me with this?

Reply:
Try this:

```js
  static isImageSupported_(minImage) {
    return new Promise(async (resolve) => {
      const imageElement = /** @type {HTMLImageElement} */(new Image());
      imageElement.src = minImage;
      if ('decode' in imageElement) {
        try {
          await imageElement.decode();
          resolve(true);
        } catch (error) {
          resolve(false);
        }
      } else {
        imageElement.onload = imageElement.onerror = () => {
          resolve(imageElement.height === 2);
        };
      }
    });
  }
```

Reply:
New error:
```
/Users/alvaro.velad/workspace/pullrequest-shaka-player/lib/util/stream_utils.js
  798:24  error  Promise executor functions should not be async  no-async-promise-executor
```

@joeyparrish , Should use a eslint-disable?

Reply:
After reading the details of why this is not recommended (https://eslint.org/docs/rules/no-async-promise-executor), I think you can go back to what you had before.  I thought we had a linter rule against .then(), but apparently we don't, so I shouldn't be fighting to remove it.

Reply:
Ready for review